### PR TITLE
testcase/le_tc/network : add ITCs for network APIs

### PR DESCRIPTION
**Pull request by arvinmittal**
Scenarios ITCs perform below tests:
itc_net_close_p_multi_times : open socket and close file descriptor multiple times
itc_net_close_n_reclose : open socket and close file descriptor and then reclose file descriptor
itc_net_dup_p_multiple_time : open socket and clone a file descriptor to arbitrary descriptor multiple times
itc_net_dup2_p_multiple_time : open socket twice to get 2 socket descriptor and clone a file descriptor to specific descriptor multiple times
itc_net_fcntl_p_dupfd : open socket and call fcntl with flag F_DUPFD
itc_net_fcntl_p_dupfd_nonblock : open socket and call fcntl with flag O_NONBLOCK and F_DUPFD
itc_net_listen_p_multiple_time : create socket, bind it and then listen and repeat multiple times
itc_net_listen_p_different_socket : create different sockets, bind it and then listen
itc_net_listen_p_after_socket_close : create socket, bind, close socket and then listen

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by manoharahk** [comment]
Commit message need to explain about each change in the commit
Sample commit message format:
Short summary of changes(Less than 50 characters would be good) Ex: network : add testcases
<Empty line>
<Detailed explanation of your change in a paragraph/multiple paragraph>
<It is good address below points>
	- Why is this change necessary?
	- How does it address the issue?
	- What side effects does this change have?
<Empty line>
<Signed-off-by>

**Comment by manoharahk** [comment]
Commit message need to explain about each change in the commit
Short (50 chars or less) summary of changes Ex: network : add testcases
Empty line
Detailed explanation of your change in a paragraph/multiple paragraph>
It is good address below points
	- Why is this change necessary?
	- How does it address the issue?
	- What side effects does this change have?
Empty line
Signed-off-by

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/le_tc/network/network_tc_main.c [hunk 1]
```diff
@@ -113,6 +113,18 @@ int network_tc_main(int argc, char *argv[])
 #ifdef CONFIG_TC_NET_DUP
 	net_dup_main();
 #endif
+#ifdef CONFIG_ITC_NET_CLOSE
+	itc_net_close_main();
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/network_tc_main.c]: Let's add these on tc_internal.h. This causes implicit function declaration.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@arvinmittal Is there any conflict? Why do you update yours to latest? As I said before, due to ```merge branch 'master' into xx```, I can't merge this pr.

**Comment by seinfra** [comment]
Code Rule Check Result:
os/arch/arm/src/armv7-r/arm_assert.c:282: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by arvinmittal** [comment]
I am cancelling this PR and raising it to resolve conflict.

**Pull request closed without merging**